modules/notification/action/action.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package action
+
+import (
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/notification/base"
+)
+
+type actionNotifier struct {
+	base.NullNotifier
+}
+
+var (
+	_ base.Notifier = &actionNotifier{}
+)
+
+// NewNotifier create a new actionNotifier notifier
+func NewNotifier() base.Notifier {
+	return &actionNotifier{}
+}
+
+func (a *actionNotifier) NotifyRenameRepository(doer *models.User, repo *models.Repository, oldName string) {
```

Comment:
Seems weird that this is the only action in `modules/notification`. I guess this is the only one called from outside models, but I wonder why the action isn't simply created by `models.ChangeRepositoryName()`.

Reply:
@guillep2k That will result an import recycle. The dependency cycle is `services` > `notifications` > `models`.

I will move more `actions` to this notifications on continue refactor PRs.